src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/StudentController.java

```diff
@@ -0,0 +1,107 @@
+package com.github.maleksandrowicz93.cqrsdemo.student;
+
+import com.github.maleksandrowicz93.cqrsdemo.student.api.StudentApi;
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.SaveStudentRequest;
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.StudentDto;
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.StudentIdentification;
+import com.github.maleksandrowicz93.cqrsdemo.student.result.ResultCode;
+import com.github.maleksandrowicz93.cqrsdemo.student.result.ResultProperty;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@FieldDefaults(makeFinal = true)
+@RequiredArgsConstructor
+class StudentController implements StudentApi {
+
+    StudentQueryFacade studentQueryFacade;
+    StudentWriteFacade studentWriteFacade;
+
+    @Override
+    public ResponseEntity<List<StudentIdentification>> findAllStudents(Integer page, Integer size) {
+        List<StudentIdentification> students = studentQueryFacade.getAllStudents(page, size);
+        return ResponseEntity.ok(students);
+    }
+
+    @Override
+    public ResponseEntity<StudentDto> addStudent(SaveStudentRequest saveStudentRequest) {
+        var result = studentWriteFacade.addStudent(saveStudentRequest);
+        return result.value()
+                .map(student -> ResponseEntity
+                        .created(URI.create(getLocation(student.id())))
+                        .body(student))
+                .orElseGet(() -> {
+                    if (ResultCode.INVALID_CREDENTIALS.equals(result.code())) {
+                        return ResponseEntity.badRequest().build();
+                    }
+                    if (ResultCode.STUDENT_ALREADY_EXISTS.equals(result.code())) {
+                        var id = UUID.fromString(result.property(ResultProperty.CONFLICTED_ID));
```

Comment:
To mapowanie UUID <-> String wygląda na wyciekającą logikę. Dwie klasy na różnych poziomach muszą wiedzieć o tym mapowaniu - nieprzyjemne i mało elastyczne API.